shared/src/main/scala/squants/mass/MomentOfIntertia.scala

```diff
@@ -0,0 +1,73 @@
+package squants.mass
+
+import squants.motion.{AngularAcceleration, NewtonMeters, Torque}
+import squants.space.{Feet, Meters}
+import squants.{AbstractQuantityNumeric, Dimension, Length, PrimaryUnit, SiBaseUnit, StrictlyPositiveQuantity, UnitConverter, UnitOfMeasure}
+
+/**
+  *
+  * @author paxelord
+  * @since 0.6
+  *
+  * @param value Double
+  */
+final class MomentOfInertia private (val value: Double, val unit: MomentOfInertiaUnit)
+    extends StrictlyPositiveQuantity[MomentOfInertia](value){
+
+  def dimension = MomentOfInertia
+
+  def toKilogramsMetersSquared = to(KilogramsMetersSquared)
+  def toPoundsSquareFeet = to(PoundsSquareFeet)
+
+  def *(angularAcceleration: AngularAcceleration): Torque = {
+    val radiansPerSecondSquared = angularAcceleration.toRadiansPerSecondSquared
+
+    NewtonMeters(toKilogramsMetersSquared * radiansPerSecondSquared)
+  }
+
+  /**
+    * For a point mass with the given MomentOfInertia rotating with a center of
+    * rotation at the given radius, return the mass of the point mass
+    * @param radius distance to axis of rotation
+    * @return mass of point mass with given radius and MomentOfInertia
+    */
+  def atCenter(radius: Length): Mass = {
+    Kilograms(toKilogramsMetersSquared / radius.squared.toSquareMeters)
+  }
+}
+
+object MomentOfInertia extends Dimension[MomentOfInertia] {
+  private[mass] def apply[A](n: A, unit: MomentOfInertiaUnit)(implicit num: Numeric[A]) = new MomentOfInertia(num.toDouble(n), unit)
```

Comment:
On a side note, I am starting to second guess whether this class should even exist. Maybe Squants should only provide compile time safety of units, and not tell people how to represent quantities, even if they are doing unusual things like having negative `Mass` or `MomentOfInertia`